frmts/vrt/pixelfunctions.cpp

```diff
@@ -2394,23 +2409,43 @@ static CPLErr MinOrMaxPixelFunc(double dfK, void **papoSources, int nSources,
                     if (bPropagateNoData)
                     {
                         dfRes = dfNoData;
+                        dfResSrc = std::numeric_limits<double>::quiet_NaN();
```

Comment:
for consistency with a few lines below, should be done within a "if constexpr (std::is_same_v<ReturnType, ReturnIndex>)"

Reply:
done